### PR TITLE
[DATA-668] carto simplify mutex in PaintMap

### PR DESCRIPTION
**Pull request by JohnN193**
ticket: https://viam.atlassian.net/browse/DATA-668

simplifies PaintMap function to use one mutex lock and copy all required data for painting the map beforehand, to ensure no mismatch occurs in the map. Also removes the painted trajectories from the output.

### CODE REVIEW

File: slam-libraries/viam-cartographer/src/slam_service/slam_service.cc [hunk 1]
```diff
@@ -178,10 +178,24 @@ std::string SLAMServiceImpl::PaintMap(bool pose_marker_flag) {
...
+    std::map<cartographer::mapping::SubmapId,
+             cartographer::mapping::proto::SubmapQuery::Response>
+        response_protos;
+
     {
         std::lock_guard<std::mutex> lk(map_builder_mutex);
         submap_poses =
             map_builder.map_builder_->pose_graph()->GetAllSubmapPoses();
+
+        for (const auto &submap_id_pose : submap_poses) {
```
Comment (kkufieta) [inline slam-libraries/viam-cartographer/src/slam_service/slam_service.cc:190]: [nit] `const auto &&submap_id_pose`

File: slam-libraries/viam-cartographer/src/slam_service/slam_service.cc [hunk 2]
```diff
@@ -192,21 +206,12 @@ std::string SLAMServiceImpl::PaintMap(bool pose_marker_flag) {
     }
 
     for (const auto &submap_id_pose : submap_poses) {
```
Comment (kkufieta) [inline slam-libraries/viam-cartographer/src/slam_service/slam_service.cc:208]: [nit] `const auto &&`

File: slam-libraries/viam-cartographer/src/slam_service/slam_service.cc [hunk 3]
```diff
@@ -192,21 +206,12 @@ std::string SLAMServiceImpl::PaintMap(bool pose_marker_flag) {
...
-            }
-        }
-
         auto submap_textures =
             absl::make_unique<::cartographer::io::SubmapTextures>();
-        submap_textures->version = response_proto.submap_version();
-        for (const auto &texture_proto : response_proto.textures()) {
+        submap_textures->version =
+            response_protos[submap_id_pose.id].submap_version();
+        for (const auto &texture_proto :
```
Comment (kkufieta) [inline slam-libraries/viam-cartographer/src/slam_service/slam_service.cc:213]: [nit] `const auto &&`

**Review by kkufieta — APPROVED** [review]
Nice work! I like the solution you came up with. 

LGTM mod a few nits

**Review by jeremyhyde-viam — APPROVED** [review]
LGTM, universal reference 4 THE WIN

File: slam-libraries/viam-cartographer/src/slam_service/slam_service.cc (hunk 3 above)
Comment (JohnN193) [reply slam-libraries/viam-cartographer/src/slam_service/slam_service.cc:213]: for whatever reason this one was throwing an error, so changed it back to a single &

**Pull request merged**